katas/args/rubiana/args.js

```diff
@@ -0,0 +1,117 @@
+//-l -p 8080 -d /usr/logs
+//-l (logging=boolean)  -p (port=number)  -d (directory=string)
+
+export class schemaClass {
+    constructor() {
+        this.flagsSchema = [{
+            flag: 'l',
+            name: 'logging',
+            type: Boolean,
+            defaultValue: false
+        },
+        {
+            flag: 'p',
+            name: 'port',
+            type: Number,
+            defaultValue: 0,
+        },
+        {
+            flag: 'd',
+            name: 'directory',
+            type: String,
+            defaultValue: ''
+        }
+        ];
+    }
+
+    splitInputStr(inputStr) {
+        let arrInput = inputStr.split(' ');
+        return arrInput;
+    }
+
+    isFlag(str) {
+        if (str.substring(0, 1) == '-' && isNaN(str.substring(1, 2))) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    validFlag(flag) {
+        if (this.flagIndex(flag) > -1) {
+            return true;
+        } else {
+            return 'Invalid! The Flag does not exist.';
+        }
+    }
+
+    flagIndex(flag) {
+        var flagIndex;
```

Comment:
If you are expecting a integer result from this function, then lets ensure that you are always returning an string regardless of the result.

So here, apparently, you are going to return `undefined` in case that flag is not found in the schema. Instead, you can set `-1` by default and that's gonna be the result in case that flag is not found